app/serializers/articles/show_serializer.rb

```diff
@@ -1,3 +1,19 @@
 class Articles::ShowSerializer < ActiveModel::Serializer
+  include ActionView::Helpers::TextHelper
+
   attributes :id, :title, :body
-end
\ No newline at end of file
+
+  def body
+    if current_user != nil 
+      if current_user.subscriber? || current_user.journalist? || current_user.publisher?
+      object.body
+      else
+        truncate(object.body, length: 350)
+      end
+    else
+      truncate(object.body, length: 350)
+    end
+  end
```

Comment:
Could be rewritten as 
```
if current_user != nil && current_user.role != nil
  object.body
else
  truncate(object.body, length: 350)
end
```
Which may or mat not be easier to read (it's less explicit who we're allowing the full content). Up to you.

Reply:
fixed